### ISSUE TITLE
Add DistanceRiemersma

### ISSUE THREAD

**Issue by makeworld-the-better-one** [lucasb-eyer/go-colorful#52]
Thiadmer Riemersma developed an algorithm that calculates color distance, with similar results to CIELUV, but it operates directly on RGB. This makes it fast and accurate.

https://www.compuphase.com/cmetric.htm

**Comment by makeworld-the-better-one** [comment]
@compuphase I'm hoping you could answer some questions about the implementation of this, seeing as you're active on GitHub.

To be clear, I am referring to this part of the article specifically:

![image](https://user-images.githubusercontent.com/25111343/106084211-e45fd980-60eb-11eb-97ca-804c0e745f22.png)


- Are the RGB input values linear or non-linear? It'd be nice if this could be specified on the website as well
- If my RGB values were in the range [0, 1] instead of [0, 255], how would the algorithm change?
	- Should the 2+ and +4 parts of the algorithm be scaled as well, becoming 4/255 and 2/255?
	- What about the parts where the value is divided by 256? If it was divided by 255 I'd be fine to just leave it out, but I'm unsure about this.

**Comment by compuphase** [comment]
Hello everyone,

I wrote the article that was quoted, so allow me to answer.

The RGB values are supposed to be non-linear (so gamma-corrected). Thank
you for pointing out that the article does not mention this. I will look
into it.

At the time that the article was written, a divide was still a costly
operation (in CPU cycles). There were still PCs in the market without FPU,
so floating point arithmetic was emulated in software (which was very
slow). What we did in our software, was to use integer arithmetic and
postpone the divide. So the part "(2 + r_avg / 256) * dR * dR" would
actually be written as "((512 + r_avg) * dR * dR) / 256" and then the
divide by 256 was replaced by a shift by 8 bits.

In short, technically a divide by 255 is more accurate, and these days
there is no need for approximative integer arithmetic anymore. If the input
RGB values are in the range [0..1], the equation would become:

    dC = sqrt( (2 + r_avg) * dR * dR  +  4 * dG * dG  +  (2 + (1 - r_avg))
* dB * dB) )

The constants 2 and 4 are not scaled. These are weight factors.

Hope this helps,
Thiadmer Riemersma

**Comment by makeworld-the-better-one** [comment]
Thank you! It's pretty interesting how you had to work around PCs without FPUs. It'd be cool to have that on the website too, as well as the more modern version of the algorithm.

I've added this to the repo and it will be in the next release.
